docs(core): Update inject error documentation (#47020)
angular/angular: aio/content/errors/NG0203.md

@name `inject()` must be called from an injection context
@category runtime

> @shortDescription `inject()` must be called from an injection context (a constructor, a factory function, a field initializer, or a function used with `EnvironmentInjector#runInContext`)

Reviewer: super nit: should we consider making it a bit less "strict" and list possible contexts as examples, not an exhaustive list? We may expand the support for `inject` function in other places in the future (like Router guards) and this content might get outdated.
```suggestion
@shortDescription `inject()` must be called from an injection context, such as a constructor, a factory function, a field initializer, or a function called from the `EnvironmentInjector#runInContext` callback.
```

Author: "or a function called from the `EnvironmentInjector#runInContext` callback"
and 
"a function used with `EnvironmentInjector#runInContext`"

are saying the same thing, in my opinion. Router guards would be "used with `EnvironmentInjector#runInContext`" under the hood and I don't think changing the wording makes it less strict. I have, however updated the sentence to remove the parentheses.

Reviewer: Thanks Andrew 👍
